### PR TITLE
vtest

### PR DESCRIPTION
**Pull request by aeschweik**
A tool for vrecord users to test their equipment without having to play a test tape. vtest should pipe test streams and files from the computer out to the monitor via the same BlackMagic card. Would welcome testing to make sure it works on other setups!

### CODE REVIEW

**Comment by kieranjol** [comment]
I haven't tested, but this sounds like a brilliant idea!

File: vtest [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+        fi
+    fi
+}
+
+_config_check(){
+    if [[ -f "${CONFIG_FILE}" ]] ; then
+        . "${CONFIG_FILE}"
+    else
+        touch "${CONFIG_FILE}"
+        exec "${SCRIPTNAME}" -e
```
Comment (retokromer) [inline vtest]: On my end, I need
```
exec "${SCRIPTDIR}/${SCRIPTNAME}" -e
```

File: vtest [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+${SCRIPTNAME} ${VERSION}
+
+${SCRIPTNAME} will pipe test files and output from the computer out to your
+monitor, via a BlackMagic capture card. It is useful for testing equipment
+without having to play a test tape.
+
+Dependencies: amiaopensource/amiaos/ffmpegdecklink --with-sdl2
+  --with-freetype --with-openjpeg
+
+Usage: ${SCRIPTNAME} [ -n | -p | -l | -e | -f | -h ] optionaltestfile
```
Comment (retokromer) [inline vtest]: `-a` is missing

File: vtest [hunk 3]
```diff
@@ -0,0 +1,454 @@
...
+DURATION.option = 1
+DURATION.option = 5
+DURATION.option = 10
+# Add a cancel button with default label
+cb.type=cancelbutton
+"
+
+# list of selections for each vplay option
+VIDEO_SOURCE_OPTIONS=("testsrc" "testsrc2" "yuvtestsrc" "rgbtestsrc" "quit")
+STANDARD_OPTIONS=("SD: NTSC (720x486, 30000/1001 fps)" "SD: PAL (720x576, 25000/1000 fps)" "HD: 1080p 25 fps (1920x1080, 25000/1000 fps)" "HD: 1080p 29.97 fps (1920x1080, 30000/1001 fps)" "HD: 1080p 30 fps (1920x1080, 30000/1000 fps)" "HD: 1080p 50 fps (1920x1080, 50000/1000 fps)" "HD: 1080i 50 fps (1920x1080, 25000/1000 fps)" "HD: 1080p 59.94 fps (1920x1080, 60000/1001 fps)" "HD: 1080i 59.94 fps (1920x1080, 30000/1001 fps)" "HD: 1080p 60 fps (1920x1080, 60000/1001 fps)")
```
Comment (retokromer) [inline vtest]: add `quit` option

**Comment by aeschweik** [comment]
Updated with those changes! Thanks for taking a look.

**Comment by retokromer** [comment]
Can this be merged? To me it’s fine.

**Pull request merged**